classes/local/courseformat/simplesection_base.php

```diff
@@ -51,14 +51,13 @@ private static function get_my_format() {
         return $format;
     }
 
-    public static function find_section_by_slug($courseid, $sectionslug) {
-        global $DB;
-
-        $sections = $DB->get_records('course_sections', ['course' => $courseid]);
-        foreach ($sections as $section) {
-            $slug = clean_moodle_url::sluggify($section->name, false);
+    public static function find_section_by_slug($course, $sectionslug) {
+        $cminfo = get_fast_modinfo($course);
+        foreach (array_keys($cminfo->sections) as $sectionnum) {
```

Comment:
It can be, and it will return null:

https://github.com/brendanheywood/moodle-local_cleanurls/blob/f9dae0f0e5887a5e669350c7768c82d397fe3aaa/classes/local/courseformat/simplesection_base.php#L64

If there is no section, it will try to see if there is an activity in the 'root' (invisible) section to use.